datadog/resource_datadog_dashboard.go

```diff
@@ -4641,11 +4706,134 @@ func buildTerraformQueryTableRequests(datadogQueryTableRequests *[]datadogV1.Tab
+		datadogQueryTableTextFormat[j] = *datadogTextFormatRule
+	}
+	return &datadogQueryTableTextFormat
+}
+func buildTerraformQueryTableTextFormat(datadogQueryTableTextFormat *[]datadogV1.TableWidgetTextFormatRule) *[]map[string]interface{} {
+	terraformQueryTableTextFormat := make([]map[string]interface{}, len(*datadogQueryTableTextFormat))
+	for i, datadogQueryTableTextFormatRule := range *datadogQueryTableTextFormat {
+		terraformQueryTableTextFormatRule := map[string]interface{}{}
+		// Required params
+		match := make(map[string]interface{})
+		match["type"] = datadogQueryTableTextFormatRule.GetMatch().Type
+		match["value"] = datadogQueryTableTextFormatRule.GetMatch().Value
+		terraformQueryTableTextFormatRule["match"] = []interface{}{match}
+		// Optional params
+		if v, ok := datadogQueryTableTextFormatRule.GetReplaceOk(); ok {
+			if v.TableWidgetTextFormatReplaceAll != nil {
+				replace := make(map[string]interface{})
+				replace["type"] = v.TableWidgetTextFormatReplaceAll.Type
+				replace["with"] = v.TableWidgetTextFormatReplaceAll.With
+				terraformQueryTableTextFormatRule["replace"] = []interface{}{replace}
+			}
+			if v.TableWidgetTextFormatReplaceSubstring != nil {
+				replace := make(map[string]interface{})
+				replace["type"] = v.TableWidgetTextFormatReplaceSubstring.Type
+				replace["with"] = v.TableWidgetTextFormatReplaceSubstring.With
+				replace["substring"] = v.TableWidgetTextFormatReplaceSubstring.Substring
+				terraformQueryTableTextFormatRule["replace"] = []interface{}{replace}
+			}
+		}
+		if v, ok := datadogQueryTableTextFormatRule.GetPaletteOk(); ok {
+			terraformQueryTableTextFormatRule["palette"] = v
+		}
+		if v, ok := datadogQueryTableTextFormatRule.GetCustomBgColorOk(); ok {
+			terraformQueryTableTextFormatRule["custom_bg_color"] = v
+		}
+		if v, ok := datadogQueryTableTextFormatRule.GetCustomFgColorOk(); ok {
+			terraformQueryTableTextFormatRule["custom_fg_color"] = v
+		}
+		terraformQueryTableTextFormat[i] = terraformQueryTableTextFormatRule
+	}
+	return &terraformQueryTableTextFormat
+}
+func getTableWidgetTextFormatMatchSchema() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"type": {
+			Description:      "Match or compare option.",
+			Type:             schema.TypeString,
+			ValidateDiagFunc: validators.ValidateEnumValue(datadogV1.NewTableWidgetTextFormatMatchTypeFromValue),
+			Required:         true,
+		},
+		"value": {
+			Description: "Table Widget Match String.",
+			Type:        schema.TypeString,
+			Required:    true,
+		},
+	}
+}
+func getTableWidgetTextFormatReplaceSchema() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"type": {
+			Description: "Table widget text format replace all type.",
+			Type:        schema.TypeString,
+			Required:    true,
+		},
+		"with": {
+			Description: "Table Widget Match String.",
+			Type:        schema.TypeString,
+			Required:    true,
+		},
+		"substring": {
+			Description: "Text that will be replaced.",
```

Comment:
Nit. Could we specify that this can only be used if type is `substring` as well. Otherwise we get a perpetual diff

Reply:
Updated.